src/adapters/base.ts

```diff
@@ -40,6 +44,12 @@ interface IRepoAdapter {
   getBaseBranch(repo: IRepo): string;
 
   getEnvironmentVariables(repo: IRepo): Promise<IEnvironmentVariables>;
+
+  createIssue(repo: IRepo): Promise<string>;
+
+  updateIssue(repo: IRepo, issueNumber: number): Promise<void>;
+
+  closeIssue(repo: IRepo, issueNumber: number): Promise<void>;
```

Comment:
Similarly, I'm thinking this can be passed as a parameter to an issue api.

Reply:
I can merge create and update as an issue api. But for close, can i use the same issue api if the state is passed as closed in the migration script

Reply:
I think so.  My guess is that there'll be some version of the following logic in the code:

```
if(issueExists){
   if(stateIsClosed){
      // call manage state function
   }  else {
      // call update issue function
   }
} else {
   // call create issue function
}
```

Reply:
i merged update and close